plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -253,42 +272,47 @@ public static String serializeStatsAsJson(DeltaLakeFileStatistics fileStatistics
         return OBJECT_MAPPER.writeValueAsString(fileStatistics);
     }
 
-    public static List<ColumnMetadata> extractSchema(MetadataEntry metadataEntry, TypeManager typeManager)
+    public static List<DeltaLakeColumnMetadata> extractSchema(MetadataEntry metadataEntry, TypeManager typeManager)
     {
+        ColumnMappingMode mappingMode = getColumnMappingMode(metadataEntry);
         return Optional.ofNullable(metadataEntry.getSchemaString())
-                .map(json -> getColumnMetadata(json, typeManager))
+                .map(json -> getColumnMetadata(json, typeManager, mappingMode))
                 .orElseThrow(() -> new IllegalStateException("Serialized schema not found in transaction log for " + metadataEntry.getName()));
     }
 
     @VisibleForTesting
-    static List<ColumnMetadata> getColumnMetadata(String json, TypeManager typeManager)
+    static List<DeltaLakeColumnMetadata> getColumnMetadata(String json, TypeManager typeManager, ColumnMappingMode mappingMode)
     {
         try {
             return stream(OBJECT_MAPPER.readTree(json).get("fields").elements())
-                    .map(node -> mapColumn(typeManager, node))
+                    .map(node -> mapColumn(typeManager, node, mappingMode))
                     .collect(toImmutableList());
         }
         catch (JsonProcessingException e) {
             throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA, getLocation(e), "Failed to parse serialized schema: " + json, e);
         }
     }
 
-    private static ColumnMetadata mapColumn(TypeManager typeManager, JsonNode node)
+    private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNode node, ColumnMappingMode mappingMode)
     {
         String fieldName = node.get("name").asText();
         JsonNode typeNode = node.get("type");
         boolean nullable = node.get("nullable").asBoolean();
-        return ColumnMetadata.builder()
+        Type columnType = buildType(typeManager, typeNode, false);
+        Optional<Type> physicalColumnType = mappingMode == ColumnMappingMode.NAME ? Optional.of(buildType(typeManager, typeNode, true)) : Optional.empty();
+        Optional<String> physicalName = mappingMode == ColumnMappingMode.NAME ? Optional.of(node.get("metadata").get("delta.columnMapping.physicalName").asText()) : Optional.empty();
```

Comment:
This treats `none` and `unknown` the same, but unknown should probably throw an unsupported exception